lib/spack/spack/cmd/pkg.py

```diff
@@ -201,14 +212,49 @@ def pkg_changed(args):
         colify(sorted(packages))
 
 
+def pkg_source(args):
+    """dump source code for a package"""
+    specs = spack.cmd.parse_specs(args.spec, concretize=False)
+    if len(specs) != 1:
+        tty.die("spack pkg source requires exactly one spec")
+
+    spec = specs[0]
+
+    # regular source dump -- just get the package and print its contents
+    if not args.canonical:
```

Comment:
Recommend flipping the logical condition here since it can sometimes be confusing to read `if not x` when both sides of `if`/`else` are used. This might look like:

```python
    if args.canonical:
        content = ph.canonical_source(spec)
        sys.stdout.write(content)
        return

    # regular source dump -- just get the package and print its contents
    filename = spack.repo.patch.filename_for_package_name(spec.name)
    if sys.stdout.isatty():
        tty.msg("%s:" % filename)
    with open(filename) as f:
        sys.stdout.write(f.read())
```

However, it's also not clear to me why we would only `tty.msg()` the `filename` when we do a non-`canonical` source dump. To remove some of this logic, I might recommend the following: 

```python
    filename = spack.repo.path.filename_for_package_name(spec.name)
    if sys.stdout.isatty():
        tty.msg("%s:" % filename)

    if args.canonical:
        content = ph.canonical_source(spec)
    else:
        with open(filename) as f:
            content = f.read() # we might have an llnl.util.filesystem method for this

    sys.stdout.write(content)
```

Reply:
ok did this.